src/MissionManager/SimpleMissionItem.cc

```diff
@@ -537,6 +537,14 @@ bool SimpleMissionItem::isLoiterItem() const
 
 bool SimpleMissionItem::showLoiterRadius() const
 {
+    // Special case: APM fixed-wing doesn't support specifying a radius with Loiter (time)
+    if (_controllerVehicle->apmFirmware() &&
```

Comment:
Ok, so give no radius is correct. It would be much better to look up in the mission command tree whether the current firmware has param3 available or not. You can get to the root of the tree with this method::
```
    const MissionCommandUIInfo *getUIInfo(Vehicle *vehicle, QGCMAVLink::VehicleClass_t vtolMode, MAV_CMD command);
```

Then you can figure out whether param3 is there or not. This way this code only needs to change once to this and then it will track whatever changes happen in the mission json across firmwares and changes which happen in taht json as newer firmwares support new things.